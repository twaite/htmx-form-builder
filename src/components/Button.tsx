import clsx from 'clsx';
import { Html } from '@elysiajs/html';

type Props = {
  class?: string;
  type?: 'button' | 'submit' | 'reset';
  variant?: 'primary' | 'secondary';
};

export default function Button(props: Html.PropsWithChildren<Props>) {
  return (
    <button
      data-testid="button"
      // Forward other props so htmx works
      {...props}
      class={clsx(
        'rounded px-3 py-2',
        {
          'bg-indigo-600 text-white dark:bg-emerald-500 ':
            !props.variant || props.variant === 'primary',
          'bg-gray-300 dark:bg-emerald-300 dark:text-slate-800':
            props.variant === 'secondary',
        },
        props.class,
      )}
      type={props.type}
    >
      {props.children}
    </button>
  );
}
