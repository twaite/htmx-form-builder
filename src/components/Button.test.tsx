import { describe, expect, test } from 'bun:test';
import Button from './Button';

describe('Button', () => {
  test('should render a button', () => {
    expect(Button({ type: 'button' })).toMatchSnapshot();
  });

  test('should render a submit button', () => {
    const button = Button({ type: 'submit' });
    document.body.innerHTML = button;

    expect(button).toMatchSnapshot();
    expect(document.querySelector('button').type).toEqual('submit');
  });
});
