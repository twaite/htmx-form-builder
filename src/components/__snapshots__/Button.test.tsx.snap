// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a button 1`] = `"<button data-testid="button" type="button" class="rounded px-3 py-2 bg-indigo-600 text-white dark:bg-emerald-500 "></button>"`;

exports[`Button should render a submit button 1`] = `"<button data-testid="button" type="submit" class="rounded px-3 py-2 bg-indigo-600 text-white dark:bg-emerald-500 "></button>"`;
